lib/exceptions.js

```diff
@@ -1,16 +1,80 @@
+export const MDBE_COMPATIBILITY_ERROR = 'MongoCompatibilityError';
+export const MDBE_CURSOR_EXHAUSTED_ERROR = 'MongoCursorExhaustedError';
+export const MDBE_CURSOR_IN_USE_ERROR = 'MongoCursorInUseError';
+export const MDBE_DECOMPRESSION_ERROR = 'MongoDecompressionError';
+export const MDBE_DRIVER_ERROR = 'MongoDriverError';
+export const MDBE_EXPIRED_SESSION_ERROR = 'MongoExpiredSessionError';
+export const MDBE_GRID_FS_CHUNK_ERROR = 'MongoGridFSChunkError';
+export const MDBE_GRID_FS_STREAM_ERROR = 'MongoGridFSStreamError';
+export const MDBE_INVALID_ARGUMENT_ERROR = 'MongoInvalidArgumentError';
+export const MDBE_KERBEROS_ERROR = 'MongoKerberosError';
+export const MDBE_MISSING_CREDENTIALS_ERROR = 'MongoMissingCredentialsError';
+export const MDBE_MISSING_DEPENDENCY_ERROR = 'MongoMissingDependencyError';
+export const MDBE_NETWORK_ERROR = 'MongoNetworkError';
+export const MDBE_NETWORK_TIMEOUT_ERROR = 'MongoNetworkTimeoutError';
+export const MDBE_NOT_CONNECTED_ERROR = 'MongoNotConnectedError';
+export const MDBE_PARSE_ERROR = 'MongoParseError';
+export const MDBE_RUNTIME_ERROR = 'MongoRunTimeError';
+export const MDBE_SERVER_CLOSED_ERROR = 'MongoServerClosedError';
+export const MDBE_SERVER_SELECTION_ERROR = 'MongoServerSelectionError';
+export const MDBE_SYSTEM_ERROR = 'MongoSystemError';
+export const MDBE_TAILABLE_CURSOR_ERROR = 'MongoTailableCursorError';
+export const MDBE_TOPOLOGY_CLOSED_ERROR = 'MongoTopologyClosedError';
+export const MDBE_TRANSACTION_ERROR = 'MongoTransactionError';
+export const MDBE_UNEXPECTED_SERVER_RESPONSE_ERROR =
+  'MongoUnexpectedServerResponseError';
+export const MDBE_SERVER_ERROR = 'MongoServerError';
+export const MDBE_WRITE_CONCERN_ERROR = 'MongoWriteConcernError';
+// Errors outside of the Mongo class of Errors
 export const WRITE_CONCERN_ERROR = 'WriteConcernError';
-export const MDBE_ERRORS = [
+export const WRITE_ERROR = 'WriteError';
+// Set of all MDBE related errors
+export const MDBE_ERRORS = new Set([
   MDBE_ERROR,
-  WRITE_ERROR,
-  BULK_WRITE_ERROR,
-  WRITE_CONCERN_ERROR
-];
+  MDBE_API_ERROR,
+  MDBE_AWS_ERROR,
+  MDBE_BATCH_REEXECUTION_ERROR,
+  MDBE_BULK_WRITE_ERROR,
+  MDBE_CHANGE_STREAM_ERROR,
+  MDBE_COMPATIBILITY_ERROR,
+  MDBE_CURSOR_EXHAUSTED_ERROR,
+  MDBE_CURSOR_IN_USE_ERROR,
+  MDBE_DECOMPRESSION_ERROR,
+  MDBE_DRIVER_ERROR,
+  MDBE_EXPIRED_SESSION_ERROR,
+  MDBE_GRID_FS_CHUNK_ERROR,
+  MDBE_GRID_FS_STREAM_ERROR,
+  MDBE_INVALID_ARGUMENT_ERROR,
+  MDBE_KERBEROS_ERROR,
+  MDBE_MISSING_CREDENTIALS_ERROR,
+  MDBE_MISSING_DEPENDENCY_ERROR,
+  MDBE_NETWORK_ERROR,
+  MDBE_NETWORK_TIMEOUT_ERROR,
+  MDBE_NOT_CONNECTED_ERROR,
+  MDBE_PARSE_ERROR,
+  MDBE_RUNTIME_ERROR,
+  MDBE_SERVER_CLOSED_ERROR,
+  MDBE_SERVER_SELECTION_ERROR,
+  MDBE_SYSTEM_ERROR,
+  MDBE_TAILABLE_CURSOR_ERROR,
+  MDBE_TOPOLOGY_CLOSED_ERROR,
+  MDBE_TRANSACTION_ERROR,
+  MDBE_UNEXPECTED_SERVER_RESPONSE_ERROR,
+  MDBE_SERVER_ERROR,
+  MDBE_WRITE_CONCERN_ERROR,
+  WRITE_CONCERN_ERROR,
+  WRITE_ERROR
+]);
```

Comment:
-1 to adding all these, we should only add what we need (and we haven't needed any of these) -- otherwise it's just more to maintain and test.

Reply:
That's a good point. There might be an automated way of adding these though. I'll look into it and if not cut them down to something more manageable. I also think not all of these errors are root errors, many of them are the cause of a `MongoError` I believe.

Reply:
ok so I trimmed this down to just the two errors that are kind of sort of thrown by the driver that are not `MongoError`s. Then I added a new function `assertMongoError` here: https://github.com/digitalbazaar/bedrock-mongodb/pull/95/commits/43ab9b0f1c3503dc31a8a8847e918c0c6638b70c
and here:
https://github.com/digitalbazaar/bedrock-mongodb/pull/95/commits/94bf5291c7994e72c0a7943895dbffaa6d86c69c

This method should work as the node mongo driver is now written using typescript so there are a lot of Abstract base classes and inheritance based type checking structures that ultimately result in 3 cases: either you get a `MongoError`, a `WriteError` or `WriteConcernError`. For the most part you only get `MongoError`, but documents can contain `WriteError` and `MongoError` can contain both `WriteError` or `WriteConcernError` so there might be edge cases were we're asserting on the cause of a `MongoError` or on the `writeErrors` property of `MongoError`.